test/com/arpnetworking/metrics/portal/expressions/impl/DatabaseExpressionRepositoryTest.java

```diff
@@ -0,0 +1,166 @@
+        try (Transaction transaction = Ebean.beginTransaction()) {
+            final models.ebean.Expression ebeanExpression1 = TestBeanFactory.createEbeanExpression();
+            ebeanExpression1.setUuid(UUID.randomUUID());
+            Ebean.save(ebeanExpression1);
+            final models.ebean.Expression ebeanExpression2 = TestBeanFactory.createEbeanExpression();
+            ebeanExpression2.setUuid(UUID.randomUUID());
+            Ebean.save(ebeanExpression2);
+            transaction.commit();
+        }
+        Assert.assertEquals(2, exprRepo.getExpressionCount());
+    }
+
+    @Test
+    public void addOrUpdateExpressionAddCase() {
+        final UUID uuid = UUID.randomUUID();
+        Assert.assertFalse(exprRepo.get(uuid).isPresent());
+        final Expression actual = TestBeanFactory.createExpressionBuilder()
+                .setId(uuid)
+                .build();
+        exprRepo.addOrUpdateExpression(actual);
+        final Optional<Expression> expected = exprRepo.get(uuid);
+        Assert.assertTrue(expected.isPresent());
+        Assert.assertEquals(expected.get(), actual);
+    }
+
+    @Test
+    public void addOrUpdateExpressionUpdateCase() throws IOException {
+        final UUID uuid = UUID.randomUUID();
+        try (Transaction transaction = Ebean.beginTransaction()) {
+            final models.ebean.Expression ebeanExpression = TestBeanFactory.createEbeanExpression();
+            ebeanExpression.setUuid(uuid);
+            Ebean.save(ebeanExpression);
+            transaction.commit();
+        }
+        final Expression actual = TestBeanFactory.createExpressionBuilder()
+                .setId(uuid)
+                .setCluster("new-cluster")
+                .build();
+        exprRepo.addOrUpdateExpression(actual);
+        final Expression expected = exprRepo.get(uuid).get();
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test(expected = PersistenceException.class)
+    public void testThrowsExceptionWhenQueryFails() {
+        final UUID uuid = UUID.randomUUID();
+        exprRepo.addOrUpdateExpression(TestBeanFactory.createExpressionBuilder()
+                .setId(uuid)
+                .setCluster("new-cluster")
+                .build());
+        models.ebean.Expression ebeanExpression1 = Ebean.find(models.ebean.Expression.class)
+                .where()
+                .eq("uuid", uuid)
+                .findUnique();
+        models.ebean.Expression ebeanExpression2 = Ebean.find(models.ebean.Expression.class)
+                .where()
+                .eq("uuid", uuid)
+                .findUnique();
+        try (Transaction transaction = Ebean.beginTransaction()) {
+            ebeanExpression1.setCluster("new-cluster1");
+            ebeanExpression2.setCluster("new-cluster2");
+            ebeanExpression2.save();
+            ebeanExpression1.save();
+            transaction.commit();
+        } catch (final IOException e) {
+            //Do Nothing
+        }
+    }
```

Comment:
I don't see any test cases for the actual query of alerts or expressions which is the most complex part of the code.